docs installation

On Oct 3, 2017, at 07:23, gyanesh patra <pgyanesh.patra@gmail.com> wrote:

Hi,
I am trying to compile and install p4c on ubuntu 16 LTS. 
The cmake & make went fine. But "make install" fails for doxygen. The details are as below:

~/p4/p4c/build$ cmake .. -DCMAKE_BUILD_TYPE=DEBUG -DENABLE_DOCS=ON -DENABLE_P4RUNTIME_TO_PD=ON

~/p4/p4c/build$make -j4
~/p4/p4c/build$sudo make install

[ 78%] Built target p4c-bm2-ss
[ 82%] Built target p4c-ebpf
[ 85%] Built target p4test
[ 89%] Built target p4c-graphs
[ 92%] Built target gtest
[100%] Built target gtestp4c
Install the project...
-- Install configuration: "DEBUG"
-- Up-to-date: /usr/local/share/p4c/p4include
-- Up-to-date: /usr/local/share/p4c/p4include/core.p4
-- Up-to-date: /usr/local/share/p4c/p4include/psa.p4
-- Up-to-date: /usr/local/share/p4c/p4include/v1model.p4
-- Up-to-date: /usr/local/share/p4c/p4include/p4d2model.p4
CMake Error at cmake_install.cmake:56 (file):
  file INSTALL cannot find
  "/home/gyanesh/arena/p4/p4c/build/doxygen_out/html".

Makefile:93: recipe for target 'install' failed
make: *** [install] Error 1

~/p4/p4c/build$

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -363,8 +363,8 @@ if (ENABLE_DOCS AND DOXYGEN_FOUND)
             ${DOXYGEN_EXECUTABLE} ${DOXYGEN_FILE}
     DEPENDS genIR
     COMMENT "Generating documentation")
-  install (FILES ${P4C_BINARY_DIR}/doxygen_out/html
-    RESOURCE DESTINATION ${P4C_ARTIFACTS_OUTPUT_DIRECTORY}/docs)
+  install (DIRECTORY ${P4C_BINARY_DIR}/doxygen-out/html/
+    DESTINATION ${P4C_ARTIFACTS_OUTPUT_DIRECTORY}/docs)
 endif()
 
 # \TODO: packaging

